packages/grafana-ui/src/components/ColorPicker/ColorPickerInput.tsx

```diff
@@ -0,0 +1,77 @@
+import { css } from '@emotion/css';
+import React, { useState, forwardRef } from 'react';
+import { RgbaStringColorPicker } from 'react-colorful';
+import { useThrottleFn } from 'react-use';
+
+import { colorManipulator, GrafanaTheme2 } from '@grafana/data';
+
+import { useStyles2, useTheme2 } from '../../themes';
+import { ClickOutsideWrapper } from '../ClickOutsideWrapper/ClickOutsideWrapper';
+import { Props as InputProps } from '../Input/Input';
+
+import ColorInput from './ColorInput';
+
+export interface ColorPickerInputProps extends Omit<InputProps, 'value' | 'onChange'> {
+  value?: string;
+  onChange: (color: string) => void;
+  /** Format for returning the color in onChange callback, defaults to 'rgb' */
+  returnColorAs?: 'rgb' | 'hex';
+}
+
+export const ColorPickerInput = forwardRef<HTMLInputElement, ColorPickerInputProps>(
+  ({ value = '', onChange, returnColorAs = 'rgb', ...inputProps }, ref) => {
+    const [currentColor, setColor] = useState(value);
+    const [isOpen, setIsOpen] = useState(false);
+    const theme = useTheme2();
+    const styles = useStyles2(getStyles);
+
+    useThrottleFn(
+      (c) => {
+        const color = theme.visualization.getColorByName(c);
+        if (returnColorAs === 'rgb') {
+          onChange(colorManipulator.asRgbString(color));
+        } else {
+          onChange(colorManipulator.asHexString(color));
+        }
+      },
+      500,
+      [currentColor]
+    );
+
+    return (
+      <ClickOutsideWrapper onClick={() => setIsOpen(false)}>
+        <div className={styles.wrapper}>
+          {isOpen && <RgbaStringColorPicker color={currentColor} onChange={setColor} className={styles.picker} />}
```

Comment:
This look different to the one used in the Basic story under the "Custom" tab. Could we maybe share that `RgbaStringColorPicker` component and it's specific style so it is consistent between those two cases?

Reply:
Ideally we should be sharing a lot between this component and `SpectrumPalette`, with the main difference being if the palette is always visible or in a popover kind of thing. Would that be very hard?

Reply:
Tbh, this is still quite experimental component, as we only have a first use case for it, so this is just a base. But we can add some more improvements in the future.

Reply:
Since `RgbaStringColorPicker` is an external library component, I exported `SpectrumPalette` styles instead and applied them to the new picker.